build.gradle

```diff
@@ -51,7 +51,7 @@ try {
     gitPropertiesName = 'iceberg-build.properties'
     gitPropertiesResourceDir = file("${rootDir}/build")
     extProperty = 'gitProps'
-    failOnNoGitDirectory = false
+    failOnNoGitDirectory = true
```

Comment:
Will this cause build verification to fail? If we're downloading the tarball and building from that will the build fail?

Reply:
No, this actually doesn't do much. The plugin is configured in a try/catch after the Palantir version plugin. So if there is not .git directory, this code doesn't even run.

Eventually, we should see if this can be moved outside of the try/catch, which was my intent here. I think that if `generateGitProperties` is called and it can't, then we want to have a failure. That's why this introduces a new task, `buildInfo` that will use the properties file in the build root if it is present. The failure we want is when you have no `.git` directory and have no `iceberg-build.properties` file.

Reply:
I tried without the try catch and having `gitProperties` in the build file causes even simple tasks like `gradle tasks` to fail because the surrounding `gitProperties` block is not registered and has no meaning to gradle. Even if it’s not used, such as `gradle clean`.

So my observation was that the try-catch block will always be needed for any `gitProperties` block.

Reply:
I think this flag is more for the case that somebody overrides the `.git` directory and that overridden directory isn’t present (at least from my testing).

Ryan’s right that it doesn’t do much.